src/main/java/com/iexec/core/chain/ChainConfig.java

```diff
@@ -28,19 +28,19 @@
 @NoArgsConstructor
 public class ChainConfig {
 
-    @Value("${chain.id}")
+    @Value("#{blockchainAdapterService.publicChainConfig.chainId}")
     private Integer chainId;
 
-    @Value("${chain.sidechain}")
+    @Value("#{blockchainAdapterService.publicChainConfig.isSidechain()}")
     private boolean isSidechain;
 
     @Value("${chain.privateAddress}")
     private String privateChainAddress;
 
-    @Value("${chain.publicAddress}")
+    @Value("#{blockchainAdapterService.publicChainConfig.nodeAddress}")
     private String publicChainAddress;
```

Comment:
I think we don't need that value anymore

Reply:
Good point, thanks!